pkg/services/serviceaccounts/manager/service.go

```diff
@@ -59,3 +59,9 @@ func (sa *ServiceAccountsService) DeleteServiceAccount(ctx context.Context, orgI
 	}
 	return sa.store.DeleteServiceAccount(ctx, orgID, serviceAccountID)
 }
+
+func (sa *ServiceAccountsService) HasServiceAccountsMigrated(ctx context.Context, orgID int64) bool {
```

Comment:
This looks like it could potentially end up being an expensive call in the hot path. What's the idea here?

Reply:
The Idea here is that we check for feature flag of `service accounts`.

Then we check if we have performed a migration, using select on the `migration_log` table for a specific migration row for that orgID

I can provide the PR which we are working on or present it tomorrow.

Reply:
> Then we check if we have performed a migration, using select on the migration_log table for a specific migration row for that orgID

That's not a query that's worth running for every page load, imo.

Reply:
initiated a discussion for this https://github.com/grafana/grafana-enterprise/discussions/2557